x-pack/legacy/plugins/code/model/commit.ts

```diff
@@ -5,7 +5,7 @@
  */
 
 export interface CommitInfo {
-  updated: Date;
+  updated: string;
```

Comment:
Since these come in as JSON, I'm pretty sure they're never going to be Date objects and instead will always be date strings, but please check my math on that.

Reply:
you are right, but javascript should be able to handle this implicitly. checkout here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON. This is also the reason why the commit index failed because of the malformat of the commit date.

Reply:
Huh, I didn't know that! I'll revert that type change.

Reply:
@mw-ding on second thought: if these are coming through as strings (albeit via `Date.toJSON()` is it still correct to type them as `Date`s? They won't have the interface of a `Date` object, and so may lead to runtime errors that Typescript can't catch. If we want to type them as `Date`, I think we would want to pass them through the `Date` constructor on the client side.

Reply:
More succinctly: this is a string that represents a date, but AFAIK Typescript only gives us either `string` or `Date`, and while `string` is less specific, it is more correct than `Date` (unless we actually convert them to Date objects)

Reply:
How about this. The `CommitInfo` is used directly for the git API, so let's change it to `string`. While for the `date` in `Commit`, let's keep it as `Date`? And we can explicitly convert it to `Date` in here: https://github.com/elastic/kibana/blob/9ff6d86379822b0ece43c58a70da5924b264b6b7/x-pack/legacy/plugins/code/server/indexer/commit_indexer.ts#L117

Reply:
Good call! Will do.

Reply:
0eb23baef42cb67fb05acd94b5e743583f29685b